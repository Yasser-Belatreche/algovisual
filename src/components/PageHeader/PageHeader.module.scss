.container {
  @apply flex items-center justify-center py-32 pt-36;
  top: -101px;

  @media only screen and (max-width: 920px) {
    @apply py-20 pt-28;
    top: -81px;

    img {
      display: none !important;
    }
  }
}
