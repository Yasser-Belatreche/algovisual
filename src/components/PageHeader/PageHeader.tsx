import Image from "next/image";
import React from "react";

// styles
import styles from "./PageHeader.module.scss";

// components
import { BlueBackground } from "../BlueBackground/BlueBackground";
import { HeroTitle } from "../HeroTitle/HeroTitle";

interface Props {
  title: string;
  backgroundImage: StaticImageData;
}

const PageHeader: React.FC<Props> = ({ title, backgroundImage }) => {
  return (
    <BlueBackground className={styles.container}>
      <HeroTitle color="white">{title}</HeroTitle>
      <Image
        alt="background image"
        src={backgroundImage}
        layout="fill"
        objectFit="cover"
      />
    </BlueBackground>
  );
};

export { PageHeader };
