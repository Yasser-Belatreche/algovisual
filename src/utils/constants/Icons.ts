import github from "../../../public/icons/github.svg";
import arrowDown from "../../../public/icons/arrow-down.svg";
import arrowLeft from "../../../public/icons/arrow-left.svg";
import code from "../../../public/icons/code.svg";

const ICONS = {
  github,
  arrowDown,
  arrowLeft,
  code,
};

export { ICONS };
