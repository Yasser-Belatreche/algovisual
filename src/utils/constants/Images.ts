import logo from "../../../public/images/Logo.png";
import heroPerson from "../../../public/images/hero-person.png";
import laptop from "../../../public/images/laptop.png";
import analytics from "../../../public/images/analytics.png";
import heroImage from "../../../public/images/hero-image.png";

const IMAGES = {
  logo,
  heroPerson,
  laptop,
  analytics,
  heroImage,
};

export { IMAGES };
