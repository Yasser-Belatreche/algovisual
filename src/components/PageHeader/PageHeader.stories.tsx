import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { PageHeader } from "./PageHeader";
import { IMAGES } from "../../utils/constants/Images";

export default {
  title: "Global/PageHeader",
  component: PageHeader,
  argTypes: {},
} as ComponentMeta<typeof PageHeader>;

const Template: ComponentStory<typeof PageHeader> = (args) => (
  <PageHeader {...args} />
);

export const PageHeaderDefault = Template.bind({});
PageHeaderDefault.args = {
  backgroundImage: IMAGES.algorithmsHeaderBg,
  title: "Algorithms",
};
