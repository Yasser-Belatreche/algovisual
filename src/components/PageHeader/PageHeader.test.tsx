import { render } from "@testing-library/react";
import { IMAGES } from "../../utils/constants/Images";
import { PageHeader } from "./PageHeader";

describe("PageHeader Component", () => {
  it("should render correctly", () => {
    const instance = render(
      <PageHeader
        title="Algorithms"
        backgroundImage={IMAGES.algorithmsHeaderBg}
      />
    );
    expect(instance.container).toMatchSnapshot();
  });
});
