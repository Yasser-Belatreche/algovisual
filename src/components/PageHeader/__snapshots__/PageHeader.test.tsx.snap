// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Component should render correctly 1`] = `
<div>
  <div
    class="container container"
    role="container"
  >
    <h1
      class="title white undefined"
    >
      Algorithms
    </h1>
    <img
      alt=""
      layout="fill"
      objectfit="cover"
      src="test-file-stub"
    />
  </div>
</div>
`;
